pswalker/recovery.py

```diff
@@ -112,29 +114,35 @@ def homs_recovery(*, detectors, motors, goals, detector_fields, index,
     """
     # Interpret args as homs mirrors and areadetector pims
     # Take the active mirror/pim pair
+    mirror = motors[index]
     yag = detectors[index]
+    sig_threshold = 0.1
     if sim:
+        # The fake mirror should not try to return to nominal
+        # This lets us test the plan
         sig = yag.detector.stats2.centroid.x
     else:
+        # The real mirror should try to return to nominal first
+        logger.info("Try recovering to nominal first...")
+        yield from mv(mirror, mirror.nominal_position)
         sig = yag.detector.stats2.centroid.y
-    mirror = motors[index]
-    try:
-        # Try recover_threshold toward nominal position
-        center = mirror.nominal_position
-        if mirror.position < center:
-            dir_initial = 1
-        else:
-            dir_initial = -1
-    except AttributeError:
-        # If no nominal position, do the shorter move first
-        dist_to_high = abs(mirror.position - mirror.high_limit)
-        dist_to_low = abs(mirror.position - mirror.low_limit)
-        if dist_to_high < dist_to_low:
-            dir_initial = 1
+        if sig.value > sig_threshold:
+            logger.info("We have beam at the nominal position.")
+            return True
         else:
-            dir_initial = -1
+            logger.info("We do not have beam at the nominal position.")
+
+    # Do the shorter move first
+    dist_to_high = abs(mirror.position - mirror.high_limit)
+    dist_to_low = abs(mirror.position - mirror.low_limit)
```

Comment:
Limits will throw an error if you request beyond it but will be fine if you are exactly at the limit. As for how the actual motor behaves, I'm not entirely sure. Previously it just wouldn't attempt a move if it was too far beyond some acceptable range, but I don't actually know where this range is defined. Maybe this is something to ping @slacAWallace about